cpp/cmake_modules/BuildUtils.cmake

```diff
@@ -97,7 +97,30 @@ function(arrow_create_merged_static_lib output_target)
   endforeach()
 
   if(APPLE)
-    set(BUNDLE_COMMAND "libtool" "-no_warning_for_no_symbols" "-static" "-o"
+    # The apple-distributed libtool is what we want for bundling, but there is
+    # a GNU libtool that has a namecollision (and happens to be bundled with R, too).
+    # We are not compatible with GNU libtool, so we need to avoid it.
+    # TODO: use a VALIDATOR when we require cmake >= 3.25
+
+    # check in the obvious places first to find Apple's libtool
+    find_program(LIBTOOL_MACOS libtool
+                 PATHS /usr/bin /Library/Developer/CommandLineTools/usr/bin
+                 NO_DEFAULT_PATH)
+
+    # if it's not found in obvious places, check on the standard path
+    if(LIBTOOL_MACOS-NOTFOUND)
+      set(LIBTOOL_MACOS "libtool")
+    endif()
+
+    # confirm that the libtool we found is not GNU libtool
+    execute_process(COMMAND ${LIBTOOL_MACOS} -V
+                    OUTPUT_VARIABLE LIBTOOL_V_OUTPUT
+                    OUTPUT_STRIP_TRAILING_WHITESPACE)
+    if(NOT "${LIBTOOL_V_OUTPUT}" MATCHES ".*cctools-([0-9.]+).*")
+      message(FATAL_ERROR "libtool found appears to be the incompatible GNU libtool")
+    endif()
```

Comment:
We might consider raising this as a warning and proceeding anyway. If it is GNU libtool, it fails pretty clearly next. That would insulate us incase Apple's libtool has a slightly different version string or something (although with a spurious error). I'm split on if this is better or not.